torchvision/models/regnet.py

```diff
@@ -0,0 +1,708 @@
+            group_width,
+            bottleneck_multiplier,
+            se_ratio,
+        )
+        self.activation = activation_layer(inplace=True)
+
+        # The projection and transform happen in parallel,
+        # and activation is not counted with respect to depth
+
+    def forward(self, x: Tensor) -> Tensor:
+        if self.proj_block:
+            x = self.bn(self.proj(x)) + self.f(x)
+        else:
+            x = x + self.f(x)
+        return self.activation(x)
+
+
+class ResBottleneckLinearBlock(nn.Module):
+    """Residual linear bottleneck block: x + F(x), F = bottleneck transform."""
+
+    def __init__(
+        self,
+        width_in: int,
+        width_out: int,
+        stride: int,
+        norm_layer: Callable[..., nn.Module],
+        activation_layer: Callable[..., nn.Module],
+        group_width: int = 1,
+        bottleneck_multiplier: float = 4.0,
+        se_ratio: Optional[float] = None,
+    ) -> None:
+        super().__init__()
+        self.has_skip = (width_in == width_out) and (stride == 1)
+        self.f = BottleneckTransform(
+            width_in,
+            width_out,
+            stride,
+            norm_layer,
+            activation_layer,
+            group_width,
+            bottleneck_multiplier,
+            se_ratio,
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        return x + self.f(x) if self.has_skip else self.f(x)
+
+
+class AnyStage(nn.Sequential):
+    """AnyNet stage (sequence of blocks w/ the same output shape)."""
+
+    def __init__(
+        self,
+        width_in: int,
+        width_out: int,
+        stride: int,
+        depth: int,
+        block_constructor: Callable[..., nn.Module],
+        norm_layer: Callable[..., nn.Module],
+        activation_layer: Callable[..., nn.Module],
+        group_width: int,
+        bottleneck_multiplier: float,
+        se_ratio: Optional[float] = None,
+        stage_index: int = 0,
+    ) -> None:
+        super().__init__()
+
+        for i in range(depth):
+            block = block_constructor(
+                width_in if i == 0 else width_out,
+                width_out,
+                stride if i == 0 else 1,
+                norm_layer,
+                activation_layer,
+                group_width,
+                bottleneck_multiplier,
+                se_ratio,
+            )
+
+            self.add_module(f"block{stage_index}-{i}", block)
+
+
+class RegNetParams:
+    def __init__(
+        self,
+        depth: int,
+        w_0: int,
+        w_a: float,
+        w_m: float,
+        group_width: int,
+        bottleneck_multiplier: float = 1.0,
+        stem_type: Callable[..., nn.Module] = SimpleStemIN,
+        stem_width: int = 32,
+        block_type: Callable[..., nn.Module] = ResBottleneckBlock,
+        activation: Callable[..., nn.Module] = nn.ReLU,
+        use_se: bool = True,
+        se_ratio: float = 0.25,
+        bn_epsilon: float = 1e-05,
+        bn_momentum: float = 0.1,
```

Comment:
Shall we pass the norm_layer also here? (see [this](https://github.com/pytorch/vision/blob/main/torchvision/models/resnet.py#L155-L160)). This is useful because it will allow us to pass other layers (such as [FrozenBN](https://github.com/pytorch/vision/blob/af228fa89bd4386816607c9d5de77f1304316983/torchvision/models/detection/backbone_utils.py#L52)) in detection.

Reply:
Sounds good.